src/bit-systems/object-menu.ts

```diff
@@ -142,13 +232,29 @@ function updateVisibility(world: HubsWorld, menu: EntityID, frozen: boolean) {
 }
 
 const hoveredQuery = defineQuery([HoveredRemoteRight]);
+const heldQuery = defineQuery([HeldRemoteRight]);
+const heldEnterQuery = enterQuery(heldQuery);
+const heldExitQuery = exitQuery(heldQuery);
 export function objectMenuSystem(world: HubsWorld, sceneIsFrozen: boolean, hubChannel: HubChannel) {
   const menu = anyEntityWith(world, ObjectMenu)!;
 
   ObjectMenu.targetRef[menu] = objectMenuTarget(world, menu, sceneIsFrozen);
   if (ObjectMenu.targetRef[menu]) {
-    moveToTarget(world, menu);
     handleClicks(world, menu, hubChannel);
+
+    heldExitQuery(world).forEach(eid => {
```

Comment:
My first thought would have been to do `hasComponent(world, Held, ObjectMenu.rotateButtonRef[menuEid)` comparison to the current transform system state.

Using quires is fine, and maybe its nice that the previous frame state is handled by the enter/exit queries but just noting. Not sure which I like better.

Reply:
I can understand.

Using `hasComponent(world, Held, ObjectMenu.rotateButtonRef[menuEid])` may be more efficient because `Held` component may be a common Component so `heldQuery` might include a way more entity ids than needed here.

But it required to manage previous frame state. I want to avoid to manage a previous frame state in a system for simplicity.

Using queries may be inefficient but we don't need to manually manage previous frame state so it is simpler.

I want to go as is for now. We may revisit later if we face performance problem.